application-analytics-ui/src/main/resources/Analytics/Code/Macros/MostViewedPages.xml

```diff
@@ -0,0 +1,820 @@
+        <classType>com.xpn.xwiki.objects.classes.NumberClass</classType>
+      </priority>
+      <supportsInlineMode>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>supportsInlineMode</name>
+        <number>5</number>
+        <prettyName>Supports inline mode</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </supportsInlineMode>
+      <visibility>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>visibility</name>
+        <number>6</number>
+        <prettyName>Macro visibility</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator>|</separator>
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity}}
+#set ($discard = $xwiki.jsx.use('Analytics.Code.Macros.MostViewedPages'))
+#set ($object =  $xwiki.getDocument("Analytics.Code.Configuration").getObject('Analytics.Code.ConfigurationClass'))
+#set ($ip = $object.getProperty("requestAddress").value)
+#set ($idSite = $object.getProperty("idSite").value)
+#set ($authToken = $object.getProperty("authToken").value)
+#set ($dashbord =  $doc.getObject('Analytics.Code.AnalyticsDashboardClass'))
+#set ($period = 'range')
+## A dashboard will have a date inside and that date will take precedence over the date set by the macro.
+#if ($dashbord)
+  #set ($startDate =  $dashbord.getProperty("startDate"))
+  #set ($endTimeDate =  $dashbord.getProperty("endDate"))
+#else
+  #set ($startDate = $xcontext.macro.params.startDate.toString())
+  #set ($endTimeDate = $xcontext.macro.params.endDate.toString())
+#end
+#set ($rows = $xcontext.macro.params.numberOfRows)
+#set ($startDateObj = $datetool.toDate('yyyy/MM/dd HH:MM', $startDate))
+#set ($startDate = $datetool.format('yyyy-MM-dd', $startDateObj))
+#set ($endDateObj = $datetool.toDate('yyyy/MM/dd HH:MM', $endDate))
+#set ($endDate = $datetool.format('yyyy-MM-dd', $endDateObj))
+#set ($date = $startDate + ',' + $endDate)
+## This will check if either the startDate or the endDate is missing, and replace the missing time interval with
+## '2007-01-01' and the current date. The date '2007-01-01' is the oldest one accepted by Matomo.
+# if(!$startDate || !$endTimeDate)
+  #set($period = 'year')
+  #set($date = 'last1')
+#end
```

Comment:
I don't understand, since your comment does not match what you actually do in code.
You should check if `endDate` is not defined and then replace it with the current date. The same for `startDate`, that should be replaced with `2007-01-01` in case is not defined

Reply:
"I forgot to make this change; it should have been:
#if(!$startTime || !$endTime)
  #set($date = '2007-01-01' + ',' + $datetool.get('yyyy-MM-dd'))
  $date
#end